[DECLUTTER WHY] Rework Privacy in Bitcoin (#569)
zkSNACKs/WasabiDoc: docs/why-wasabi/TransactionGraph.md

---
{
  "title": "Transaction Graph",
  "description": "On how Bitcoin transactions are interconnected, how this is dangereous for privacy, and how to fix it. This is the Wasabi documentation, an archive of knowledge about the open-source, non-custodial and privacy-focused Bitcoin wallet for desktop."
}
---

# Transaction Graph

## Problem

_**Public transaction history**_

Because of the input and output model of Bitcoin, there is a chain of digital signatures all the way from the coinbase reward, to the current UTXO.
This transaction history can reveal sensitive information of the spending patterns of individuals.
The receiver of a coin can look back into the transaction history of the sender.
And the sender can see the future spending of the receiver.

## Wasabi's Solution

_**Zero Link CoinJoins**_

In order to obfuscate the link between outputs and inputs, Wasabi uses the [Zero Link](https://github.com/nopara73/zerolink) CoinJoin protocol.

> The Wasabi central coordinator cannot steal and cannot spy, he simply helps many peers to build a huge transaction, with many inputs, and many outputs.

Reviewer: ```suggestion
The Wasabi central coordinator cannot steal and cannot spy, it simply helps many peers to build a huge transaction, with many inputs, and many outputs.
```

The non-private inputs can be linked to their previous transaction history.

> However, the equal value CoinJoin outputs with an anonymity set can not be tied to the inputs.

Reviewer: ```suggestion
However, the equal value CoinJoin outputs with an anonymity set cannot be tied to the inputs.
```
